lib/elixir/lib/module/checker.ex

```diff
@@ -3,53 +3,65 @@ defmodule Module.Checker do
 
   @moduledoc false
 
-  def verify(module_map, binary, cache) do
-    warnings = warnings(module_map, cache)
-    binary = chunk(binary, module_map)
+  def verify(map, chunks, cache) do
+    # TODO: Add ExCk chunk to state so that ParallelChecker does not
+    #       have to load from disk again
+
+    warnings = warnings(map, cache)
+    binary = add_chunk(chunks, map)
     {binary, warnings}
   end
 
-  defp chunk(binary, module_map) do
-    checker_chunk = build_chunk(module_map)
-    {:ok, _module, all_chunks} = :beam_lib.all_chunks(binary)
-    {:ok, binary} = :beam_lib.build_module([checker_chunk | all_chunks])
+  defp add_chunk(chunks, map) do
+    checker_chunk = build_chunk(map)
+    chunks = [checker_chunk | :lists.keydelete('ExCk', 1, chunks)]
```

Comment:
This is only necessary for runtime modules, correct?

Reply:
Yes, compiled modules will not have the chunk. Do you want to keep them separate?